codalab/server/bundle_manager.py

```diff
@@ -307,11 +307,14 @@ def _bring_offline_stuck_running_bundles(self, workers):
                 logger.info('Bringing bundle offline %s: %s', bundle.uuid, failure_message)
                 self._model.transition_bundle_worker_offline(bundle)
 
-    def _schedule_run_bundles_on_workers(self, workers, staged_bundles_to_run):
+    def _schedule_run_bundles_on_workers(self, workers, staged_bundles_to_run, user_info_cache):
         """
-        Schedules STAGED bundles to run on the given workers. If user_owned is
-        True, then schedules on workers run by the owner of each bundle.
-        Otherwise, uses CodaLab-owned workers, which have user ID root_user_id.
+        Schedules STAGED bundles to run on the given workers. Always tries to schedule bundles to
+        run on workers that are owned by the owner of each bundle first. If there are no such qualified
+        private workers, uses CodaLab-owned workers, which have user ID root_user_id.
+        :param workers: a WorkerInfoAccessor object containing worker related information e.g. running uuid
+        :param staged_bundles_to_run: a list of tuples each contains a valid bundle and its bundle resources
+        :param user_info_cache: a dictionary storing the mapping from user id to user information
```

Comment:
Simplify: "dictionary mapping user id to user information"

Reply:
Updated